### ISSUE TITLE
Aptly fails to parse large Binary Packages Indices

### ISSUE THREAD

**Issue by grodr** [aptly-dev/aptly#859]
Aptly fails with `ERROR: unable to update: bufio.Scanner: token too long` when running `aptly mirror update` on a mirror for a repository with large Binary Packages Indices `Packages.gz` file.

To fully reproduce this error one can do the following:
1. Add http://ddebs.ubuntu.com/ as a mirror (specifically the `universe` component has a large Packages.gz index)
`aptly mirror create ddebs http://ddebs.ubuntu.com/ bionic universe`

2. Update the mirror
`aptly mirror update ddebs`

This produces the following output
```
Downloading http://ddebs.ubuntu.com/dists/bionic/InRelease...
Downloading http://ddebs.ubuntu.com/dists/bionic/Release...
Downloading http://ddebs.ubuntu.com/dists/bionic/Release.gpg...
gpgv: Signature made Fri Apr 27 02:53:24 2018 UTC
gpgv:                using RSA key C8CAB6595FDFF622
gpgv: Good signature from "Ubuntu Debug Symbol Archive Automatic Signing Key (2016) <ubuntu-archive@lists.ubuntu.com>"
Downloading & parsing package files...
Downloading http://ddebs.ubuntu.com/dists/bionic/universe/binary-amd64/Packages.gz...
ERROR: unable to update: bufio.Scanner: token too long
```

I believe this error is happening in `aptly/deb/remote.go` somewhere in the `DownloadPackageIndexes` function (due to the presence of `Downloading & parsing package files...` message in the output).

Searching for the error leads me to believe this is a failure with bufio Scanner attempting to read a file that is too large (e.g. https://www.socketloop.com/tutorials/golang-read-large-file-with-bufio-scanner-cause-token-too-long-error). Suggested fixes are to use bufio.Reader.ReadLine instead.

This is running on Ubuntu 18.04 with `aptly 1.2.0-3`

This prevents the mirroring of any repo with a large `Packages.gz` file.

**Comment by smira** [comment]
I believe this shouldn't be related to the file size, but rather to the line length most probably. As 'token' is a line in this case

**Comment by smira** [comment]
But I'll take a look at fixing this nevertheless.

**Comment by grodr** [comment]
Awesome, thanks! Let me know if any additional details would be helpful.

**Comment by smira** [comment]
oh yeah,

```
$ wc -L Packages 
66758 Packages
```

it has a line which is longer than 64k

**Comment by smira** [comment]
@grodr oh, you're using really old aptly (1.2.0). This has been already fixed, you can try latest version 1.4.0 (www.aptly.info/download).

**Comment by grodr** [comment]
Updating fixed the issue, thanks!
